Library/Homebrew/formula.rb

```diff
@@ -2393,8 +2292,87 @@ def to_hash
   end
 
   # @private
-  def to_hash_with_variations
-    hash = to_hash
+  def to_api_hash
+    api_hash = {
+      "desc"                 => desc,
+      "license"              => SPDX.license_expression_to_string(license),
+      "homepage"             => homepage,
+      "urls"                 => urls_hash.transform_values(&:compact),
+      "post_install_defined" => post_install_defined?,
+      "ruby_source_path"     => ruby_source_path,
+      "ruby_source_checksum" => { "sha256" => ruby_source_checksum&.hexdigest },
+    }
+
+    dep_hash = dependencies_hash
+               .except("recommended_dependencies", "optional_dependencies")
```

Comment:
I'll hold off on making more changes here because of @Bo98's proposal to serialize dependencies differently.

- https://github.com/Homebrew/brew/pull/16541#discussion_r1468911967